atc/lidar/scanner.go

```diff
@@ -70,26 +72,7 @@ func (s *scanner) scanResources(ctx context.Context, resources []db.Resource, re
 	waitGroup.Wait()
 }
 
-func (s *scanner) scanResourceTypes(ctx context.Context, resourceTypes db.ResourceTypes) {
-	logger := lagerctx.FromContext(ctx)
-	waitGroup := new(sync.WaitGroup)
-	for _, resourceType := range resourceTypes {
-		waitGroup.Add(1)
-		go func(resourceType db.ResourceType, resourceTypes db.ResourceTypes) {
-			defer func() {
-				err := util.DumpPanic(recover(), "scanning resource type %d", resourceType.ID())
-				if err != nil {
-					logger.Error("panic-in-scanner-run", err)
-				}
-			}()
-			defer waitGroup.Done()
-			s.check(ctx, resourceType, resourceTypes)
-		}(resourceType, resourceTypes)
-	}
-	waitGroup.Wait()
-}
-
-func (s *scanner) check(ctx context.Context, checkable db.Checkable, resourceTypes db.ResourceTypes) {
+func (s *scanner) check(ctx context.Context, checkable db.Checkable, resourceTypesMap map[int]db.ResourceTypes) {
```

Comment:
I agree with @aoldershaw 's comment, empty array is just nil. So here we just just pass in `resourceTypes db.ResourceType` rather than a map.